src/wp-includes/option.php

```diff
@@ -1536,48 +1489,11 @@ function add_network_option( $network_id, $option, $value ) {
 	 */
 	$value = apply_filters( "pre_add_site_option_{$option}", $value, $option, $network_id );
 
-	$notoptions_key = "$network_id:notoptions";
-
 	if ( ! is_multisite() ) {
 		$result = add_option( $option, $value, '', 'no' );
 	} else {
-		$cache_key = "$network_id:$option";
-
-		// Make sure the option doesn't already exist.
-		// We can check the 'notoptions' cache before we ask for a DB query.
-		$notoptions = wp_cache_get( $notoptions_key, 'site-options' );
-
-		if ( ! is_array( $notoptions ) || ! isset( $notoptions[ $option ] ) ) {
-			if ( false !== get_network_option( $network_id, $option, false ) ) {
-				return false;
-			}
-		}
-
-		$value = sanitize_option( $option, $value );
-
-		$serialized_value = maybe_serialize( $value );
-		$result           = $wpdb->insert(
-			$wpdb->sitemeta,
-			array(
-				'site_id'    => $network_id,
-				'meta_key'   => $option,
-				'meta_value' => $serialized_value,
-			)
-		);
-
-		if ( ! $result ) {
-			return false;
-		}
-
-		wp_cache_set( $cache_key, $value, 'site-options' );
-
-		// This option exists now.
-		$notoptions = wp_cache_get( $notoptions_key, 'site-options' ); // Yes, again... we need it to be fresh.
-
-		if ( is_array( $notoptions ) && isset( $notoptions[ $option ] ) ) {
-			unset( $notoptions[ $option ] );
-			wp_cache_set( $notoptions_key, $notoptions, 'site-options' );
-		}
+		$value  = sanitize_option( $option, $value );
+		$result = add_metadata( 'site', $network_id, $option, $value, true );
```

Comment:
```suggestion
		$result = add_metadata( 'site', $network_id, wp_slash( $option ), wp_slash( $value ), true );
```

ref: https://github.com/WordPress/wordpress-develop/blob/deafd1193c244acfae80bb4507de4d8b532fa9e5/src/wp-includes/meta.php#L52-L53

Reply:
Great call out. I have added unit tests and this fix. https://github.com/WordPress/wordpress-develop/pull/2478/commits/faf732ada8e0f3092ffabb3e7d013b778dfade39